### PR TITLE
EXP-30421 Detokenizer: API documentation and swagger json

### PR DESCRIPTION
**Pull request by pandianr-dev**
@sarraloew Please review

Detokenizer: API documentation and swagger json

### CODE REVIEW

File: src/api-reference/Detokenizer/Detokenizer.swagger2.json [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    "application/json",
+    "application/xml"
+  ],
+  "consumes": [
+    "application/json",
+    "application/xml"
+  ],
+  "info": {
+    "title": "Detokenizer Service",
+    "description": "Detokenizer Service allows clients to retrieve user's credit card number from Concur Expense in a secure way.",
```
Comment (MarcBell-SAP) [inline src/api-reference/Detokenizer/Detokenizer.swagger2.json]:
```suggestion
    "description": "The Detokenizer Service allows clients to retrieve the user's credit card number from Concur Expense in a secure way. The Detokenizer service returns the user's credit card number encrypted with a public key that the client provides in the request. The client will be able to decrypt the user's credit card number using their private key.",
```

File: src/api-reference/Detokenizer/Detokenizer.swagger2.json [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    }
+  ],
+  "paths": {
+    "/company/{companyUUID}/creditcard/{creditcardGUID}": {
+      "get": {
+        "tags": [
+          "Resources"
+        ],
+        "summary": "Get credit card account details",
+        "description": "Gets the encrypted credit card account number",
```
Comment (MarcBell-SAP) [inline src/api-reference/Detokenizer/Detokenizer.swagger2.json]:
```suggestion
        "description": "Returns the credit card number encrypted with the public key provided in the request.",
```

File: src/api-reference/Detokenizer/Detokenizer.swagger2.json [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            "name": "creditcardGUID",
+            "in": "path",
+            "description": "Credit card GUID",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "pubkeyAlgorithm",
+            "in": "query",
+            "description": "RSA Algorithm used by PublicKey for card number encryption.",
```
Comment (MarcBell-SAP) [inline src/api-reference/Detokenizer/Detokenizer.swagger2.json]:
```suggestion
            "description": "The RSA Algorithm used by the PublicKey for credit card number encryption.",
```

File: src/api-reference/Detokenizer/Detokenizer.swagger2.json [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "pubkeyFormat",
+            "in": "query",
+            "description": "Format used by Public key",
```
Comment (MarcBell-SAP) [inline src/api-reference/Detokenizer/Detokenizer.swagger2.json]:
```suggestion
            "description": "Public key format",
```

File: src/api-reference/Detokenizer/Detokenizer.markdown [hunk 5]
```diff
@@ -0,0 +1,108 @@
+---
+ ## Detokenizer Service
+
+---
+
+{% include prerelease.html %}
+
+Detokenizer Service allows clients to retrieve user's credit card number from Concur Expense in a secure way. Detokenizer service returns the card number encrypted with a public key that the client provides in the request. Client will be able to decrypt the card number using their private key.
```
Comment (MarcBell-SAP) [inline src/api-reference/Detokenizer/Detokenizer.markdown]:
```suggestion
The Detokenizer Service allows clients to retrieve the user's credit card number from Concur Expense in a secure way. The Detokenizer service returns the user's credit card number encrypted with a public key that the client provides in the request. The client will be able to decrypt the user's credit card number using their private key.
```

File: src/api-reference/Detokenizer/Detokenizer.markdown [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+## <a name="products-editions"></a>Products and Editions
+
+* Concur Expense Professional Edition
+
+## <a name="scope-usage"></a>Scope Usage
+
+Name|Description|Endpoint
+---|---|---
+`creditcardaccountnumber.read or creditcardaccount.read`|Reads credit card data from concur expense|/detokenizer/v1/company/{company}/creditcard/{creditcard}
```
Comment (MarcBell-SAP) [inline src/api-reference/Detokenizer/Detokenizer.markdown]:
```suggestion
`creditcardaccountnumber.read or creditcardaccount.read`|Reads credit card data from Concur Expense|/detokenizer/v1/company/{companyUUID}/creditcard/{creditcardGUID} 
```

File: src/api-reference/Detokenizer/Detokenizer.markdown [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+---|---|---
+`creditcardaccountnumber.read or creditcardaccount.read`|Reads credit card data from concur expense|/detokenizer/v1/company/{company}/creditcard/{creditcard} 
+
+## <a name="access-token-usage"></a>Access Token Usage
+
+This API requires a Company level JWT token
+
+## <a name="get-credit-card-account-details"></a>Get credit card account details
+
+Returns the credit card number encrypted with the public key provided in the request
```
Comment (MarcBell-SAP) [inline src/api-reference/Detokenizer/Detokenizer.markdown]:
```suggestion
Returns the credit card number encrypted with the public key provided in the request.
```

File: src/api-reference/Detokenizer/Detokenizer.markdown [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+
+```shell
+GET https://{region}.api.concursolutions.com/detokenizer/v1/company/{companyUUID}/creditcard/{creditcardGUID}
+```
+
+##### Parameters
+
+Name|Type|Format|Description
+---|---|---|---
+`pubkeyAlgorithm `|`string`|query|RSA Algorithm used by PublicKey for card number encryption
```
Comment (MarcBell-SAP) [inline src/api-reference/Detokenizer/Detokenizer.markdown]:
```suggestion
`pubkeyAlgorithm `|`string`|query|The RSA Algorithm used by the PublicKey for credit card number encryption.
```

File: src/api-reference/Detokenizer/Detokenizer.markdown [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+`pubkeyFormat  `|`string`|query |Format used by Publickey
```
Comment (MarcBell-SAP) [inline src/api-reference/Detokenizer/Detokenizer.markdown]:
```suggestion
`pubkeyFormat  `|`string`|query |Public key format
```

File: src/api-reference/Detokenizer/Detokenizer.markdown [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+`pubkey  `|`string`|query|Public key
+`company  `|`string`|path |Company UUID
```
Comment (MarcBell-SAP) [inline src/api-reference/Detokenizer/Detokenizer.markdown]:
```suggestion
`companyUUID  `|`string`|path |Company UUID
```

File: src/api-reference/Detokenizer/Detokenizer.markdown [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+`creditcard  `|`string`|path |Credit card GUID
```
Comment (MarcBell-SAP) [inline src/api-reference/Detokenizer/Detokenizer.markdown]:
```suggestion
`creditcardGUID  `|`string`|path |Credit card GUID
```

**Review by MarcBell-SAP — CHANGES_REQUESTED** [review]
Please review my suggestions. These are general grammar and consistency corrections.

**Comment by pandianr-dev** [comment]
@MarcBell-SAP Addressed all your suggestions.

**Review by MarcBell-SAP — APPROVED** [review]
Approved. Merge the pull request.

**Pull request merged**